SingularityExecutor/src/main/java/com/hubspot/singularity/executor/task/SingularityExecutorTaskProcessCallable.java

```diff
@@ -79,12 +79,12 @@ private boolean runHealthcheck() {
 
       try {
         Retryer<Boolean> retryer = RetryerBuilder.<Boolean>newBuilder()
-            .retryIfResult(bool -> !bool)
+            .retryIfResult(bool -> bool)
```

Comment:
if the process is not alive though, don't we want to have this whole method return false (i.e. did not pass health check). This is the place for the enum/ternary return type I mentioned. We need to differentiate waiting vs process dead vs succeeded when deciding if we need to retry